tests/test_router_bgp.rb

```diff
@@ -229,6 +229,27 @@ def test_get_asnum
     bgp.destroy
   end
 
+  def test_nsr
+    nsr(setup_default)
+    nsr(setup_vrf)
+  end
+
+  def nsr(bgp)
+    if platform == :nexus || !@vrf[/default/]
+      assert_raises(Cisco::UnsupportedError) do
+        bgp.nsr = true
+      end
+    else
```

Comment:
Add a check to make sure the default is correct before you set anything directly:

```ruby
else
  assert_equal(bgp.nsr, bgp.default_nsr)
  bgp.nsr = true
  ...
```